packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -143,6 +143,38 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
     });
   }, [schemaFields, serverVersion]);
 
+  const onFocus = (editorView: EditorEventCallbackContext, content: string) => {
+    if (hasAutofix && content === '') {
+      editorView.setContent('{}', 1);
+      onChange('{}');
+    }
+  };
+
+  const onReplace = (
```

Comment:
This probably should account for partial replacing if we really want to handle this case. Was this a requirement for this ticket?

Reply:
Partial replacing is a corner case, it's not considered in the use case, that is why it's only for full replace.

Reply:
Then the callback name should be changed so that it's obvious to the users of this interface that this is added only for a very specific case of full replace, or, way more preferably, this should not be exposed like that from the editor component at all.

Also, just do double-check again: is this really a case we are supposed to handle? I don't remember this being mentioned anywhere in the discussions for this ticket. Selecting anything in the editor should be a signal that user knows what they are doing. The case we are trying to solve is somebody who doesn't know that query should have brackets around it when starting to type + adjust possible initial paste because this bracket auto-insert we are adding can brake it